src/Uuid/ModelUuid.php

```diff
@@ -87,32 +116,24 @@ public function id(): string
 
 		// overwrite the content in memory and in the content file;
 		// use the most basic write method to avoid object cloning
-		$this->model->content()->update($data);
-		$this->model->writeContent($data);
+		$model->content('default')->update($data);
+		$model->writeContent($data, 'default');
+
+		// update the default translation
+		if ($kirby->multilang() === true) {
+			$model->translation('default')->update($data);
+		}
 
+		// switch back to the original user
 		$kirby->impersonate($user);
 
 		// TODO: replace the above in 3.9.0 with
 		// App::instance()->impersonate(
 		// 	'kirby',
-		// 	fn () => $this->model = $this->model->writeContent($data)
+		// 	fn () => $model->writeContent($data, $languageCode)
```

Comment:
```suggestion
		// 	fn () => $model->writeContent($data, 'default')
```

However I'm wondering if we even need impersonation at all (now that we use `writeContent()`, which doesn't check permissions). Removing all of the impersonation stuff would make the code much simpler.

Reply:
Uhhh, I haven't thought about that. You are totally right. I'll give it a try.